src/java/za/co/mip/ablduck/utilities/CommentParser.java

```diff
@@ -0,0 +1,328 @@
+/**
+ * Copyright 2017 MIP Holdings
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+package za.co.mip.ablduck.utilities;
+
+import org.apache.tools.ant.Task;
+import org.commonmark.node.Node;
+import org.commonmark.parser.Parser;
+import org.commonmark.renderer.html.HtmlRenderer;
+
+import za.co.mip.ablduck.javadoc.Javadoc;
+
+import java.io.IOException;
+import java.util.Objects;
+import java.util.regex.Pattern;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+
+public class CommentParser {
+    private String comment;
+    private String source;
+    private Map<String, String> nestedComments = new HashMap<>();
+    private Javadoc javadocParser;
+    private Pattern oldCommentDetectPattern = Pattern.compile("^\\s*?\\/\\*\\s*?-+.+$",
+            Pattern.MULTILINE);
+    private String oldCommentTokenPattern = "(^\\s*?[TOKEN]\\s*?:([\\s\\S]*?))(?:\\n|\\z)*(?:^\\s*?Component\\s*?:|^\\s*?Author\\s*?:|^\\s*?File\\s*?:|^\\s*?Purpose\\s*?:|^\\s*?Syntax\\s*?:|^\\s*?Description\\s*?:|^\\s*?Author\\(s\\)\\s*?:|^\\s*?Created\\s*?:|^\\s*?Notes\\s*?:|^\\s*?@param|^\\s*?@return|\\z)";
+    private Map<String, Pattern> oldCommentPatterns = new HashMap<>();
+
+    public CommentParser(Task ablduck) {
+        javadocParser = new Javadoc(ablduck);
+
+        String[] oldCommentTokens = {"File", "Purpose", "Syntax", "Description", "Author(s)",
+                "Created", "Notes", "Author", "Component"};

```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "Notes" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3AS1192)
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "Author" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3AS1192)